testcases/PowerNVDump.py

```diff
@@ -305,6 +309,76 @@ def runTest(self):
         return boot_type
 
 
+class SBECrash_MPIPL(PowerNVDump):
+
+    def runTest(self):
+
+        if "OpenBMC" in self.bmc_type:
+            cmd = "mboxctl --reset"
+        elif "SMC" in self.bmc_type:
+            raise self.skipTest(
+                "SBE initiated MPIPL is not supported on SMC yet")
+        #run mboxctl reset on BMC
+        self.cv_BMC.run_command(cmd)
+        self.c.run_command("uname -a")
+        self.c.run_command("cat /etc/os-release")
+        #Get all chip ids on the machine
+        xscom_dirs = self.c.run_command(
+            "ls /sys/firmware/devicetree/base/ | grep xscom")
+        chip_ids = []
+        for node in xscom_dirs:
+            if node:
+                chip_id = self.c.run_command(
+                    "lsprop /sys/firmware/devicetree/base/%s/ibm,chip-id" % node)
+                chip_ids.append(chip_id[-1].strip())
+        chip_ids = sorted(chip_ids, reverse=True)
+        r_workdir = "/tmp/skiboot"
+        # Make sure /tmp/skiboot directory does not exist
+        self.c.run_command("rm -rf %s" % r_workdir)
+
+        # Clone skiboot git repository
+        r_cmd = "git clone --depth 1 https://github.com/open-power/skiboot.git %s" % r_workdir
+        self.c.run_command(r_cmd)
+        # Compile putscom utility
+        r_cmd = "cd %s/external/xscom-utils; make;cd" % r_workdir
+        self.c.run_command(r_cmd)
+        try:
+            # Check existence of putscom utility after compiling.
+            r_cmd = "test -f %s/external/xscom-utils/putscom" % r_workdir
+            self.c.run_command(r_cmd)
+        except CommandFailed as cf:
+            raise self.skipTest(
+                "putscom utility missing after compiling!!! %s" % str(cf))
+        # trigger SBE initiated MPIPL
+        for chip_id in chip_ids:
+            self.c.pty.sendline(
+                "%s/external/xscom-utils/putscom -c %s 0x50008 0x0002000000000000" % (r_workdir, chip_id))
+        done = False
```

Comment:
Please don't make assumption on primary chip ID. if we trigger on different order it may cause MPIPL to fail.

Reply:
Got it.